src/test/scala/org/broadinstitute/dsde/workbench/leonardo/auth/sam/SamAuthProviderSpec.scala

```diff
@@ -131,4 +134,66 @@ class SamAuthProviderSpec extends TestKit(ActorSystem("leonardotest")) with Free
     samAuthProvider.samClient.notebookClusters should not contain ((project, name1, userInfo.userEmail) -> Set("connect", "read_policies", "status", "delete", "sync"))
   }
 
+  "should cache hasNotebookClusterPermission results" in isolatedDbTest {
+    val samAuthProvider = getSamAuthProvider
+
+    // cache should be empty
+    samAuthProvider.notebookAuthCache.size shouldBe 0
+
+    // populate backing samClient
+    samAuthProvider.samClient.notebookClusters += (project, name1, userInfo.userEmail) -> Set("sync")
+
+    // call provider method
+    samAuthProvider.hasNotebookClusterPermission(userInfo.userEmail, SyncDataToCluster, project, name1).futureValue shouldBe true
+
+    // cache should contain 1 entry
+    samAuthProvider.notebookAuthCache.size shouldBe 1
+    val key = NotebookAuthCacheKey(userInfo.userEmail, SyncDataToCluster, project, name1, implicitly[ExecutionContext])
+    samAuthProvider.notebookAuthCache.asMap.containsKey(key) shouldBe true
+    samAuthProvider.notebookAuthCache.asMap.get(key).futureValue shouldBe true
+
+    // remove info from samClient
+    samAuthProvider.samClient.notebookClusters.remove((project, name1, userInfo.userEmail))
+
+    // provider should still return true because the info is cached
+    samAuthProvider.hasNotebookClusterPermission(userInfo.userEmail, SyncDataToCluster, project, name1).futureValue shouldBe true
+  }
+
+  "should cache canSeeAllClustersInProject results" in isolatedDbTest {
+    val samAuthProvider = getSamAuthProvider
+
+    // cache should be empty
+    samAuthProvider.notebookAuthCache.size shouldBe 0
+
+    // populate backing samClient
+    samAuthProvider.samClient.billingProjects += (project, userInfo.userEmail) -> Set("list_notebook_cluster")
+
+    // call provider method
+    samAuthProvider.canSeeAllClustersInProject(userInfo.userEmail, project).futureValue shouldBe true
+    samAuthProvider.canSeeAllClustersInProject(WorkbenchEmail("somecreep@example.com"), project).futureValue shouldBe false
```

Comment:
Could potentially put repeated things like `WorkbenchEmail("somecreep@example.com")`

Reply:
That's definitely something I should have done when I wrote the other tests in this file :)